shimmingtoolbox/cli/b0shim.py

```diff
@@ -944,6 +921,78 @@ def _initial_in_bounds(coefs, bounds):
     return list_coils
 
 
+def calculate_scanner_constraints(constraints, scanner_shim_settings, order, manufacturer):
+    """ Calculate the constraints that should be used for the scanner by considering the current shim settings and the
+        absolute bounds
+
+    Args:
+        constraints (dict): Constraints of the scanner coils
+        scanner_shim_settings (dict): Dictionary containing the shim settings of the scanner ('f0', 'order1', 'order2')
+        order (int): Order of the scanner coils (0 or 1 or 2)
+        manufacturer (str): Name of the MRI manufacturer
+
+    Returns:
+        dict: Updated constraints of the scanner
+    """
+    def _initial_in_bounds(coefs, bounds):
+        """Makes sure the initial values are within the bounds of the constraints"""
+        if len(coefs) != len(bounds):
+            raise RuntimeError("The scanner coil's bounds is not the same length as the initial bounds")
+        for i_bound in range(len(bounds)):
+            if bounds[i_bound][0] is None and bounds[i_bound][1] is None:
+                continue
+            if bounds[i_bound][1] is not None:
+                if not coefs[i_bound] <= bounds[i_bound][1]:
+                    logger.warning(f"Initial scanner coefs are outside the bounds allowed in the constraints: "
+                                   f"{bounds[i_bound]}, initial: {coefs[i_bound]}")
+            if bounds[i_bound][0] is not None:
+                if not bounds[i_bound][0] <= coefs[i_bound]:
+                    logger.warning(f"Initial scanner coefs are outside the bounds allowed in the constraints: "
+                                   f"{bounds[i_bound]}, initial: {coefs[i_bound]}")
+
+    if order == 0:
+        initial_coefs = [0]
+    elif order == 1:
```

Comment:
Can you explain, why you are doing this here ? (maybe add a quick comment to explain where these numbers came from)